core/src/test/scala/org/apache/spark/util/UtilsSuite.scala

```diff
@@ -1184,6 +1184,25 @@ class UtilsSuite extends SparkFunSuite with ResetSystemProperties with Logging {
     assert(Utils.getSimpleName(classOf[MalformedClassObject.MalformedClass]) ===
       "UtilsSuite$MalformedClassObject$MalformedClass")
   }
+
+  test("stringHalfWidth") {
+    assert(Utils.stringHalfWidth(null) == 0)
+    assert(Utils.stringHalfWidth("") == 0)
+    assert(Utils.stringHalfWidth("ab c") == 4)
+    assert(Utils.stringHalfWidth("1098") == 4)
+    assert(Utils.stringHalfWidth("mø") == 2)
+    assert(Utils.stringHalfWidth("γύρ") == 3)
```

Comment:
I think it is OK to type Unicode chars into source files because git and Scala ought to be set up to encode and read the source as UTF8. I have avoided it in the past and used Unicode escapes (but put actual string in comment) to be safe. I don't feel strongly. I think we have other instances of Unicode in the source. Just a comment.

Reply:
Do you mean that the string here needs to be converted to a unicode escape form?

Reply:
Don't change it. There are Unicode characters elsewhere in the source, and it's clearer this way.

Reply:
I found similar code in the `org.apache.commons.lang3.StringUtils` class.
So the same test was added.
In StringUtils 6934 line is comment:
`* StringUtils.isNumeric("\u0967\u0968\u0969")  = true`